frontend/tests/unit/utils.shootEditorCompletions.spec.js

```diff
@@ -0,0 +1,316 @@
+//
+// Copyright (c) 2019 by SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the 'License');
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an 'AS IS' BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+import { ShootEditorCompletions } from '@/utils/shootEditorCompletions'
+import CodeMirror from 'codemirror'
+import 'codemirror/mode/yaml/yaml.js'
+
+import repeat from 'lodash/repeat'
+
+import chai from 'chai'
+import sinon from 'sinon'
+import sinonChai from 'sinon-chai'
+chai.use(sinonChai)
+const expect = chai.expect
+
+document.body.createTextRange = function () {
```

Comment:
sandbox

Reply:
cannot use sandbox as stubbing non existant functions is not possible in sinon. However, I can remove the function after the tests in an after() hook.